lib/sql.js

```diff
@@ -690,9 +690,23 @@ SQLConnector.prototype.update = function(model, where, data, options, cb) {
  */
 SQLConnector.prototype._replace = function(model, where, data, options, cb) {
   var stmt = this.buildReplace(model, where, data, options);
-  this.execute(stmt.sql, stmt.params, options, cb);
+  this.execute(stmt.sql, stmt.params, options, function(err, info) {
+    if (info.affectedRows === 0 || err) {
```

Comment:
If `err` is set then your current code discards it and replaces with a 404 error. I think that can make troubleshooting difficult for users, because e.g. when the database connection drops, we report 404 instead of 500.

What is your reasoning for replacing `err` with 404?

I am proposing:

```js
if (err) return cb(err);
if (info.affectedRows === 0) {
  return cb(errorIdNotFoundForReplace(where.id));
}
// ...
```